influxdb3_server/src/query_executor/mod.rs

```diff
@@ -979,4 +982,232 @@ mod tests {
             assert_batches_sorted_eq!(t.expected, &batches);
         }
     }
+
+    #[test_log::test(tokio::test)]
+    async fn query_file_limits_default() {
+        let (write_buffer, query_executor, time_provider, _) = setup(None).await;
+        // Perform some writes to multiple tables
+        let db_name = "test_db";
+        // perform writes over time to generate WAL files and some snapshots
+        // the time provider is bumped to trick the system into persisting files:
+        for i in 0..1298 {
```

Comment:
I see in the comment that the time provider is bumped to trick the system into persisting files, but just so I understand correctly -- is that because the system is configured at some level to persist to the object store at some interval, ie 30 seconds or so?

Reply:
Yes exactly! Normally this would be by default I think every 10 minutes or so.

Reply:
Snapshot happens based on the number of WAL files. Incrementing the time makes it so that the writes fall into different gen1 blocks of time, each of which will result in a single parquet file.